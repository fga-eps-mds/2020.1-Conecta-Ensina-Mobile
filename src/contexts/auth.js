import AsyncStorage from '@react-native-community/async-storage';
import React, {createContext, useState, useEffect} from 'react';
import {Alert} from 'react-native';
import {HOST} from '@env';

export const AuthContext = createContext({});

export default function AuthProvider({children}) {
  const [user, setUser] = useState(null);
  const [typeUser, setTypeUser] = useState(null);
  const [teacher, setTeacher] = useState(null);
  const [student, setStudent] = useState(null);
  const [loading, setLoading] = useState(true);

  const Host = `${HOST}`;

  useEffect(() => {
    async function loadStorage() {
      const storageTypeUser = await AsyncStorage.getItem('Auth_typeUser');
      const storageUser = await AsyncStorage.getItem('Auth_user');

      if (storageTypeUser) {
        if (JSON.parse(storageTypeUser) === 1) {
          setTypeUser('Adm');
          setLoading(false);
        } else if (JSON.parse(storageTypeUser) === 2) {
          setTypeUser('Professor');
          setLoading(false);
        } else {
          setTypeUser('Aluno');
          setLoading(false);
        }
      }
      if (storageUser) {
        const dataUser = JSON.parse(storageUser);
        let usuario = {
          id: dataUser.id,
          firstName: dataUser.firstName,
          lastName: dataUser.lastName,
          email: dataUser.email,
          password: dataUser.password,
          role: dataUser.role,
          cellphone: dataUser.cellphone,
        };
        setUser(usuario);
        setLoading(false);
        if (dataUser.role == 2 || dataUser.role == 3) {
          const storageStudent = await AsyncStorage.getItem('Auth_student');
          if (storageStudent) {
            const dataStudent = JSON.parse(storageStudent);
            let estudante = {
              birthdate: dataStudent.birthdate,
              grade: dataStudent.grade,
              institution: dataStudent.school,
              cpf: dataStudent.cpf,
              cep: dataStudent.cep,
              number: dataStudent.number,
              details: dataStudent.details,
              description: dataStudent.description,
              special: dataStudent.special,
            };
            setStudent(estudante);
            setLoading(false);
          }
          if (dataUser.role == 2) {
            const storageTeacher = await AsyncStorage.getItem('Auth_teacher');
            if (storageTeacher) {
              const dataTeacher = JSON.parse(storageTeacher);
              let professor = {
                photo: dataTeacher.photo,
                video: dataTeacher.video,
                graduation_area: dataTeacher.graduation_area,
                degree: dataTeacher.degree,
                bank: dataTeacher.bank,
                agency: dataTeacher.agency,
                account: dataTeacher.account,
              };
              setTeacher(professor);
              setLoading(false);
            }
          }
        }
      }
      setLoading(false);
    }
    loadStorage();
  }, []);

  async function signIn(email, password) {
    const settings = {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        email: email,
        password: password,
      }),
    };
    const fetchResponse1 = await fetch(`${Host}/api/user/login`, settings);
    try {
      const data = await fetchResponse1.json();
      console.log('Success');
      if (data.message) {
        if (data.message === 'Login efetuado com sucesso!') {
          console.log('login efetuado');
          storageTypeUser(data.role);
          renderData(data.id, data.role, password);
          if (data.role === 1) {
            setTypeUser('Adm');
            console.log('Adm');
          } else if (data.role === 2) {
            setTypeUser('Professor');
            console.log('Professor');
          } else {
            setTypeUser('Aluno');
            console.log('aluno');
          }
          setUser(data);
        }
      }
    } catch (error) {
      console.error('Error:', error);
    }
  } //close signIn function

  async function registerAluno(values, {setStatus}) {
    var ok = false;
    const settings = {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        firstName: values.name,
        lastName: values.surname,
        email: values.email,
        password: values.password,
        cellphone: values.cellphone,
        birthdate: values.birthdate,
        grade: values.grade,
        institution: values.school,
        cpf: values.cpf,
        cep: values.cep,
        number: values.num,
        details: values.details,
        description: values.description,
        special: values.special,
      }),
    };
    const fetchResponse = await fetch(`${Host}/api/student/create`, settings);
    try {
      const data = await fetchResponse.json();
      console.log('Success');
      if (data.message) {
        if (data.message === 'Estudante criado com sucesso!') {
          ok = true;

          renderData(
            data.data.user.id,
            data.data.user.role,
            data.data.user.password,
          );
          setTypeUser('Aluno');
          storageTypeUser(data.data.user.role);
        } else if (data.message.name) {
          if (
            data.message.name === 'SequelizeUniqueConstraintError' &&
            data.message.fields.email
          ) {
            setStatus({email: 'Email j치 foi registrado'});
          } else if (
            data.message.name === 'SequelizeUniqueConstraintError' &&
            data.message.fields.cpf
          ) {
            setStatus({cpf: 'CPF j치 foi registrado'});
          }
        }
      }
    } catch (error) {
      console.error('Error:', error);
    }
    if (ok) {
      RegistroAlert(3);
    }
  } //close registerAluno fuction

  async function registerProf(values, {setStatus}) {
    var ok = false;
    console.log(
      'firstName: ' +
        values.name +
        '\n' +
        'lastName: ' +
        values.surname +
        '\n' +
        'email: ' +
        values.email +
        '\n' +
        'password: ' +
        values.password +
        '\n' +
        'cellphone: ' +
        values.cellphone +
        '\n' +
        'birthdate: ' +
        values.birthdate +
        '\n' +
        'grade: ' +
        values.grade +
        '\n' +
        'institution: ' +
        values.school +
        '\n' +
        'cpf: ' +
        values.cpf +
        '\n' +
        'cep: ' +
        values.cep +
        '\n' +
        'number: ' +
        values.num +
        '\n' +
        'details: ' +
        values.details +
        '\n' +
        'description: ' +
        values.description +
        '\n' +
        'special: ' +
        values.special +
        '\n' +
        'photo: ' +
        values.photo +
        '\n' +
        'video: ' +
        values.video +
        '\n' +
        'graduation_area: ' +
        values.graduation_area +
        '\n' +
        'degree: ' +
        values.degree +
        '\n' +
        'bank: ' +
        values.bank +
        '\n' +
        'agency: ' +
        values.agency +
        '\n' +
        'account: ' +
        values.account +
        '\n',
    );
    const settings = {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        firstName: values.name,
        lastName: values.surname,
        email: values.email,
        password: values.password,
        cellphone: values.cellphone,
        birthdate: values.birthdate,
        grade: values.grade,
        institution: values.school,
        cpf: values.cpf,
        cep: values.cep,
        number: values.num,
        details: values.details,
        description: values.description,
        special: values.special,
        photo: values.photo,
        video: values.video,
        graduation_area: values.graduation_area,
        degree: values.degree,
        bank: values.bank,
        agency: values.agency,
        account: values.account,
      }),
    };

    const fetchResponse = await fetch(`${Host}/api/teacher/create`, settings);
    try {
      const data = await fetchResponse.json();
      console.log('Success');
      if (data.message) {
        if (data.message === 'Professor criado com sucesso!') {
          ok = true;
          renderData(
            data.data.user.id,
            data.data.user.role,
            data.data.user.password,
          );
          setTypeUser('Professor');
          storageTypeUser(data.data.user.role);
        } else if (data.message.name) {
          if (
            data.message.name === 'SequelizeUniqueConstraintError' &&
            data.message.fields.email
          ) {
            setStatus({email: 'Email j치 foi registrado'});
          } else if (
            data.message.name === 'SequelizeUniqueConstraintError' &&
            data.message.fields.cpf
          ) {
            setStatus({cpf: 'CPF j치 foi registrado'});
          }
        }
      }
    } catch (error) {
      console.error('Error:', error);
    }
    if (ok) {
      RegistroAlert(2);
    }
  } //close registerProf function

  const RegistroAlert = ({role}) => {
    Alert.alert(
      'Registro',
      'Registro Concluido com sucesso',
      [
        {
          text: 'Finalizar',
          onPress: () => {
            if (role === 2) {
              setTypeUser('Professor');
            } else {
              setTypeUser('Aluno');
            }
          },
        },
      ],
      {cancelable: false},
    );
  };

  async function renderData(id, role, password) {
    var ok = false;
    try {
      let response = await fetch(`${Host}/api/user/` + id);
      const data = await response.json();
      let usuario = {
        id: id,
        firstName: data.data.user.firstName,
        lastName: data.data.user.lastName,
        email: data.data.user.email,
        password: password,
        role: data.data.user.role,
        cellphone: data.data.user.cellphone,
      };
      if (role === 2 || role === 3) {
        ok = true;
      }
      storageUser(usuario);
      setUser(usuario);
    } catch (error) {
      console.error(error);
    }
    if (ok) {
      try {
        let response = await fetch(`${Host}/api/student/` + id);
        const data = await response.json();
        let estudante = {
          birthdate: data.data.student.birthdate,
          grade: data.data.student.grade,
          institution: data.data.student.institution,
          cpf: data.data.student.cpf,
          cep: data.data.student.cep,
          number: data.data.student.number,
          details: data.data.student.details,
          description: data.data.student.description,
          special: data.data.student.special,
        };
        setStudent(estudante);
        storageStudent(estudante);
      } catch (error) {
        console.error(error);
      }
      if (role === 2) {
        try {
          let response = await fetch(`${Host}/api/teacher/` + id);
          const data = await response.json();
          let professor = {
            photo: data.data.teacher.photo,
            video: data.data.teacher.video,
            graduation_area: data.data.teacher.graduation_area,
            degree: data.data.teacher.degree,
            bank: data.data.teacher.bank,
            agency: data.data.teacher.agency,
            account: data.data.teacher.account,
          };
          setTeacher(professor);
          storageTeacher(professor);
        } catch (error) {
          console.error(error);
        }
      }
    }
  }

  async function updateUser(values, id) {
    const settings1 = {
      method: 'PUT',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        firstName: values.name,
        lastName: values.surname,
        email: values.email,
        cellphone: values.cellphone,
        password: user.password,
      }),
    };
    const fetchResponse1 = await fetch(`${Host}/api/user/${id}`, settings1);

    try {
      const data = await fetchResponse1.json();
      console.log('Success');
    } catch (error) {
      console.error('Error:', error);
    }
    if (user.role === 2 || user.role === 3) {
      const settings2 = {
        method: 'PUT',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          agentRole: user.role,
          birthdate: student.birthdate,
          grade: values.grade,
          institution: values.school,
          cpf: student.cpf,
          cep: values.cep,
          number: values.num,
          details: values.details,
          description: values.description,
          special: student.special,
        }),
      };
      const fetchResponse2 = await fetch(
        `${Host}/api/student/${id}`,
        settings2,
      );
      try {
        const data2 = await fetchResponse2.json();
        console.log('Success');
      } catch (error) {
        console.error('Error:', error);
      }
      if (user.role === 2) {
        const settings3 = {
          method: 'PUT',
          headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
          },
          body: JSON.stringify({
            agentRole: user.role,
            video: values.video,
            graduation_area: teacher.graduation_area,
            degree: values.degree,
            bank: values.bank,
            agency: values.agency,
            account: values.account,
          }),
        };
        const fetchResponse3 = await fetch(
          `${Host}/api/teacher/${id}`,
          settings3,
        );
        try {
          const data3 = await fetchResponse3.json();
          console.log('Success');
        } catch (error) {
          console.error('Error:', error);
        }
      }
    }
    renderData(user.id, user.role, user.password);
  }

  async function storageTypeUser(data) {
    await AsyncStorage.setItem('Auth_typeUser', JSON.stringify(data));
  }

  async function storageUser(data) {
    await AsyncStorage.setItem('Auth_user', JSON.stringify(data));
  }

  async function storageStudent(data) {
    await AsyncStorage.setItem('Auth_student', JSON.stringify(data));
  }

  async function storageTeacher(data) {
    await AsyncStorage.setItem('Auth_teacher', JSON.stringify(data));
  }

  async function signOut() {
    await AsyncStorage.clear().then(() => {
      setUser(null);
      setTypeUser(null);
      setStudent(null);
      setTeacher(null);
    });
  }

  return (
    <AuthContext.Provider
      value={{
        signed: !!user,
        user,
        loading,
        teacher,
        student,
        typeUser,
        Host,
        signIn,
        registerAluno,
        registerProf,
        updateUser,
        signOut,
      }}>
      {children}
    </AuthContext.Provider>
  );
}
