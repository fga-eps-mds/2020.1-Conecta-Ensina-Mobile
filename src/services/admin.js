export async function getTeacherList(Host) {
  try {
    const fetchResponse = await fetch(`${Host}/api/student/status/0`);
    const data = await fetchResponse.json();
    if (data.message === 'Nenhum professor pendente') {
      return null;
    } else {
      return data.data.student;
    }
  } catch (error) {
    return error;
  }
}

export async function getReportedUsers(Host) {
  try {
    const fetchResponse = await fetch(`${Host}/api/student/status/2`);
    const data = await fetchResponse.json();
    if (data.message === 'Nenhum professor pendente') {
      return null;
    } else {
      return data.data.student;
    }
  } catch (error) {
    return error;
  }
}

export async function getProfessoUser(Host, id) {
  try {
    console.log(id);
    const response = await fetch(`${Host}/api/student/${id}`);
    const data = await response.json();
    console.log(data);
    return data.data.student;
  } catch (error) {
    return error;
  }
}

export async function statusUpdate(Host, user, id, status) {
  const settings = {
    method: 'PUT',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      status: status,
      agentRole: user.role,
    }),
  };
  try {
    const fetchResponse1 = await fetch(
      `${Host}/api/student/status/${id}`,
      settings,
    );
    const data = await fetchResponse1.json();
    return data;
  } catch (error) {
    return error;
  }
}
