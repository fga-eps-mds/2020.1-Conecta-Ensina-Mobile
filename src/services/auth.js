export async function signIn(email, password, Host) {
  const settings = {
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      email: email,
      password: password,
    }),
  };
  const fetchResponse1 = await fetch(`${Host}/api/user/login`, settings);
  try {
    const data = await fetchResponse1.json();
    console.log('Success:', data);
    if (data.message) {
      if (data.message === 'Login efetuado com sucesso!') {
        console.log('login efetuado');
        return data;
        /*renderData(data.id, data.role);
        if (data.role === 1) {
          setTypeUser('Adm');
          console.log('Adm');
        } else if (data.role === 2) {
          setTypeUser('Professor');
          console.log('Professor');
        } else {
          setTypeUser('Aluno');
          console.log('aluno');
        }
        setUser(data);*/
      }
    }
  } catch (error) {
    console.error('Error:', error);
  }
} //close signIn function
