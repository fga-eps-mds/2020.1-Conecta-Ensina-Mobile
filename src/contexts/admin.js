import React, {createContext, useState, useContext, useEffect} from 'react';
import {AuthContext} from './auth';

export const AdmContext = createContext({});

function AdmProvider({children}) {
  const [students, setStudents] = useState(null);
  const [userDB, setUserDB] = useState(null);
  const [teacher, setTeacher] = useState(null);

  const {Host, user} = useContext(AuthContext);

  async function getProfessorList() {
    const fetchResponse = await fetch(`${Host}/api/student/status/0`);

    console.log(Host);

    try {
      const data = await fetchResponse.json();
      setStudents(data.data.student);
    } catch (error) {
      return error;
    }
  }

  async function statusUpdate(id) {
    const settings = {
      method: 'PUT',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        status: 1,
        agentRole: user.role,
      }),
    };
    const fetchResponse1 = await fetch(
      `${Host}/api/student/status/${id}`,
      settings,
    );
    try {
      const data = await fetchResponse1.json();
      console.log('Success:', data);
    } catch (error) {
      console.error('Error:', error);
    }
  }

  async function getProfessoUser(id) {
    const fetchResponse = await fetch(`${Host}/api/user/${id}`);
    try {
      const data = await fetchResponse.json();
      let usuario = {
        id: id,
        firstName: data.data.user.firstName,
        lastName: data.data.user.lastName,
        email: data.data.user.email,
        password: data.data.user.password,
        role: data.data.user.role,
        cellphone: data.data.user.cellphone,
      };
      setUserDB(usuario);
      try {
        let response = await fetch(`${Host}/api/teacher/${id}`);
        const data = await response.json();
        let professor = {
          photo: data.data.teacher.photo,
          video: data.data.teacher.video,
          graduation_area: data.data.teacher.graduation_area,
          degree: data.data.teacher.degree,
          bank: data.data.teacher.bank,
          agency: data.data.teacher.agency,
          account: data.data.teacher.account,
        };
        setTeacher(professor);
      } catch (error) {
        console.error(error);
      }
    } catch (error) {
      return error;
    }
  }

  return (
    <AdmContext.Provider
      value={{
        students,
        userDB,
        teacher,
        getProfessorList,
        statusUpdate,
        getProfessoUser,
      }}>
      {children}
    </AdmContext.Provider>
  );
}

export default AdmProvider;
