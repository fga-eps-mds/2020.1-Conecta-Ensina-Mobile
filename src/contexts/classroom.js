import React, {createContext, useState, useContext} from 'react';
import {AuthContext} from './auth';
import * as Class from '../services/classroom';
import * as Student from '../services/student';
import * as Address from '../services/findAddress';
import * as GeoCode from '../services/geocode';
import {FiltersContext} from '../contexts/filters';
import {StudentContext} from '../contexts/student';

export const ClassroomContext = createContext({});

export default function ClassroomProvider({children}) {
  const [classroom, setClassroom] = useState({});
  const [firstClass, setFirstClass] = useState(null);
  const [classes, setClasses] = useState(null);
  const {Host, user} = useContext(AuthContext);
  const {stackStudents} = useContext(StudentContext);
  const {filter} = useContext(FiltersContext);
  const [statusClasses, setStatusClasses] = useState([]);
  const [statusClass, setStatusClass] = useState({});
  const [coordenates, setCoordenates] = useState({});

  async function loadNextClass() {
    const response = await Class.getNextClassroom(Host);
    if (classroom !== response) {
      setFirstClass(response);
    }
  }
  async function createClass(teacher) {
    //console.log('Esse aqui Ã© o professor: ' + teacher);
    const student = await Student.getStudent(Host, user.id);
    const response = await Class.createClass(
      user,
      teacher,
      filter,
      student.student,
      Host,
    );
  }

  async function loadUserClasses() {
    const response = await Class.getUserClassroom(Host, user.id);
    if (classroom !== response) {
      setClassroom(response);
    }
  }

  async function getClassroom(status) {
    let teacher, student;
    if (user.role === 2) {
      student = '';
      teacher = user.id;
    } else {
      student = user.id;
      teacher = '';
    }
    const response = await Class.getClassroom(student, teacher, status, Host);

    if (classroom !== response) {
      //console.log(response);
      setStatusClasses(response);
    }
  }

  async function updateStatusClassroom(id) {
    let response = await Class.readClass(Host, id);
    //console.log(response.status);
    if (response.status === 1) {
      response = await Class.updateStatusClassroom(id, 2, Host);
      //console.log(response.message);
    } else if (response.status === 2) {
      response = await Class.updateStatusClassroom(id, 3, Host);
      //console.log(response.message);
    } else if (response.status === 3) {
      response = await Class.updateStatusClassroom(id, 4, Host);
      //console.log(response.message);
    } else if (response.status === 4) {
      response = await Class.updateStatusClassroom(id, 5, Host);
      //console.log(response.message);
    }
  }
  async function loadStatusClasses(status) {
    const response = await Class.getStatusClassroom(Host, user.id, status);
    if (statusClasses !== response) {
      await setStatusClasses(response);
    }
    //console.log(response);
    return response;
  }

  async function loadStatusClassesStudents(status) {
    const response = await loadStatusClasses(status);
    let studentList = await {list: []};
    await Promise.all(
      response.map(async (item) => {
        const student = await Student.getStudent(Host, item.student);
        studentList.list = await studentList.list.concat(student);
      }),
    );
    await stackStudents(studentList.list);
  }

  async function readClass(id) {
    const response = await Class.readClass(Host, id);
    const address = await Address.findAddress(response.cep);
    let responseClass = {
      id: response.role,
      status: response.status,
      teacher: response.teacher,
      student: response.student,
      grade: response.grade,
      subject: response.subject,
      dtclass: response.dtClass,
      duration: response.duration,
      cep: response.cep,
      number: response.number,
      details: response.details,
      address: address,
    };

    setClassroom(responseClass);
  }

  async function getClass() {
    const response = await Class.getClass(Host, user);
    setClasses(response);
  }

  async function updateStatusClasses(id, status) {
    const response = await Class.updateStatus(id, status, Host);
    if (classroom !== response) {
      setClassroom(response);
    }
  }

  async function geoCode(cep) {
    const response = await GeoCode.geocode(cep);
    if (coordenates !== response) {
      //console.log(response.results[0].geometry.location);
      setCoordenates(response.results[0].geometry.location);
    }
  }

  return (
    <ClassroomContext.Provider
      value={{
        classroom,
        firstClass,
        classes,
        loadNextClass,
        createClass,
        loadUserClasses,
        getClassroom,
        updateStatusClassroom,
        loadStatusClasses,
        readClass,
        statusClasses,
        statusClass,
        setStatusClass,
        setStatusClasses,
        Host,
        getClass,
        updateStatusClasses,
        loadStatusClassesStudents,
        geoCode,
        coordenates,
      }}>
      {children}
    </ClassroomContext.Provider>
  );
}
