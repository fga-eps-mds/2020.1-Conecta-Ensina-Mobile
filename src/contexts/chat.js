import React, {createContext, useState, useContext, useEffect} from 'react';
import {AuthContext} from './auth';
import * as Chat from '../services/chat';
import { ClassroomContext } from './classroom';

export const chatContext = createContext({});

export default function ChatProvider({children}) {
  const {Host, user} = useContext(AuthContext);
  const {classroom} = useContext(ClassroomContext);
  const [chat, setChat] = useState(null);

  useEffect(() => {
      readChat()
  }, [])

  async function createChat(text) {
    const response = await Chat.createMessage(
        user.id, 
        text, 
        classroom.id, 
        classroom.teacher, 
        classroom.student, 
        Host)
  }

  async function readChat() {
    console.log(classroom)
    const response = await Chat.getChat(Host, classroom.id);
    if (chat !== response) {
      setChat(response);
    }
  }
  return (
    <chatContext.Provider
      value={{
        readChat,
        createChat,
        chat
      }}>
      {children}
    </chatContext.Provider>
  );
}
