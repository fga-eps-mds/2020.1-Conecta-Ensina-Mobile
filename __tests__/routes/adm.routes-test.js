import React from 'react';
import {render} from '@testing-library/react-native';
import AdmRoutes from '../../src/routes/adm.routes';
import {NavigationContainer} from '@react-navigation/native';
import {AuthContext} from '../../src/contexts/auth';
import {AdmContext} from '../../src/contexts/admin';

describe('Testing AdmRoutes ', () => {
  test('Should Match Snapshot', () => {
    const route = render(
      <NavigationContainer>
        <AdmRoutes />
      </NavigationContainer>,
    ).toJSON();

    expect(route).toMatchSnapshot();
  });
});
