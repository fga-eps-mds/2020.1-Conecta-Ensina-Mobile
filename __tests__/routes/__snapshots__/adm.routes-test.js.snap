// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AdmRoutes  Should Match Snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
                "marginTop": 0,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "#184E79",
                            "display": "flex",
                            "flexBasis": 0,
                            "flexGrow": 1,
                            "flexShrink": 1,
                            "justifyContent": "flex-end",
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "flexBasis": 0,
                              "flexDirection": "row-reverse",
                              "flexGrow": 10,
                              "flexShrink": 1,
                              "marginLeft": 30,
                            },
                          ]
                        }
                      />
                      <View
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#ffffff",
                              "borderTopLeftRadius": 20,
                              "borderTopRightRadius": 20,
                              "display": "flex",
                              "flexBasis": 0,
                              "flexGrow": 78,
                              "flexShrink": 1,
                              "justifyContent": "center",
                            },
                          ]
                        }
                      >
                        <RCTScrollView
                          data={
                            Array [
                              Object {
                                "id": "101",
                                "img": Object {
                                  "testUri": "../../../src/assets/books.png",
                                },
                                "name": "Professores Pendentes",
                              },
                              Object {
                                "id": "171",
                                "img": Object {
                                  "testUri": "../../../src/assets/books.png",
                                },
                                "name": "UsuÃ¡rios Reportados",
                              },
                            ]
                          }
                          disableVirtualization={false}
                          getItem={[Function]}
                          getItemCount={[Function]}
                          horizontal={true}
                          initialNumToRender={10}
                          keyExtractor={[Function]}
                          maxToRenderPerBatch={10}
                          onContentSizeChange={[Function]}
                          onEndReachedThreshold={2}
                          onLayout={[Function]}
                          onMomentumScrollEnd={[Function]}
                          onScroll={[Function]}
                          onScrollBeginDrag={[Function]}
                          onScrollEndDrag={[Function]}
                          removeClippedSubviews={false}
                          renderItem={[Function]}
                          scrollEventThrottle={50}
                          stickyHeaderIndices={Array []}
                          style={
                            Array [
                              Object {
                                "alignSelf": "center",
                                "flexDirection": "row",
                                "paddingBottom": "120%",
                                "position": "absolute",
                              },
                            ]
                          }
                          updateCellsBatchingPeriod={50}
                          viewabilityConfigCallbackPairs={Array []}
                          windowSize={21}
                        >
                          <View>
                            <View
                              onLayout={[Function]}
                              style={
                                Array [
                                  Object {
                                    "flexDirection": "row",
                                  },
                                  null,
                                ]
                              }
                            >
                              <View
                                accessible={true}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "backgroundColor": "#D9D9D9",
                                    "borderRadius": 10,
                                    "height": 95,
                                    "marginBottom": 20,
                                    "marginRight": 20,
                                    "opacity": 1,
                                    "width": 95,
                                  }
                                }
                                testID="SquareButton"
                              >
                                <Image
                                  style={
                                    Array [
                                      Object {
                                        "height": 50,
                                        "marginTop": 8,
                                        "width": 60,
                                      },
                                    ]
                                  }
                                />
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "paddingTop": 10,
                                      },
                                    ]
                                  }
                                >
                                  <Text
                                    black={true}
                                    smaller={true}
                                    style={
                                      Array [
                                        Object {
                                          "color": "black",
                                          "fontFamily": "Roboto",
                                          "fontSize": 12,
                                          "textAlign": "center",
                                        },
                                      ]
                                    }
                                  >
                                    Professores Pendentes
                                  </Text>
                                </View>
                              </View>
                            </View>
                            <View
                              onLayout={[Function]}
                              style={
                                Array [
                                  Object {
                                    "flexDirection": "row",
                                  },
                                  null,
                                ]
                              }
                            >
                              <View
                                accessible={true}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "backgroundColor": "#D9D9D9",
                                    "borderRadius": 10,
                                    "height": 95,
                                    "marginBottom": 20,
                                    "marginRight": 20,
                                    "opacity": 1,
                                    "width": 95,
                                  }
                                }
                                testID="SquareButton"
                              >
                                <Image
                                  style={
                                    Array [
                                      Object {
                                        "height": 50,
                                        "marginTop": 8,
                                        "width": 60,
                                      },
                                    ]
                                  }
                                />
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "paddingTop": 10,
                                      },
                                    ]
                                  }
                                >
                                  <Text
                                    black={true}
                                    smaller={true}
                                    style={
                                      Array [
                                        Object {
                                          "color": "black",
                                          "fontFamily": "Roboto",
                                          "fontSize": 12,
                                          "textAlign": "center",
                                        },
                                      ]
                                    }
                                  >
                                    UsuÃ¡rios Reportados
                                  </Text>
                                </View>
                              </View>
                            </View>
                          </View>
                        </RCTScrollView>
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "flexBasis": 0,
                                "flexDirection": "column-reverse",
                                "flexGrow": 1,
                                "flexShrink": 1,
                                "marginBottom": "10%",
                              },
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "backgroundColor": "#E46270",
                                "borderRadius": 10,
                                "height": 30,
                                "justifyContent": "center",
                                "opacity": 1,
                                "width": "80%",
                              }
                            }
                            testID="signout"
                          >
                            <Text
                              bigSmall={true}
                              style={
                                Array [
                                  Object {
                                    "color": "#ffffff",
                                    "fontFamily": "Roboto",
                                    "fontSize": 16,
                                    "textAlign": "center",
                                  },
                                ]
                              }
                              white={true}
                            >
                              Sair
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "height": 64,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 1,
        }
      }
    />
  </RNCSafeAreaProvider>
</View>
`;
